### PR TITLE
[fix][broker] Fix return the earliest position when query position by timestamp.

### PR DESCRIPTION
**Pull request by hanmz**
Fixes https://github.com/apache/pulsar/issues/20368

Master Issue: https://github.com/apache/pulsar/issues/20368

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/hanmz/pulsar/pull/1

### CODE REVIEW

**Review by Jason918 — APPROVED** [review]
LGTM

**Review by wolfstudy — APPROVED** [review]
LGTM +1

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentMessageFinderTest.java [hunk 1]
```diff
@@ -243,6 +243,43 @@ public void findEntryFailed(ManagedLedgerException exception, Optional<Position>
...
+        config.setRetentionTime(1, TimeUnit.HOURS);
+        ManagedLedger ledger = factory.open(ledgerAndCursorName, config);
+        ManagedCursorImpl cursor = (ManagedCursorImpl) ledger.openCursor(ledgerAndCursorName);
+
+        ledger.addEntry(createMessageWrittenToLedger("msg1"));
+        Thread.sleep(100);
+        ledger.addEntry(createMessageWrittenToLedger("msg2"));
+        Thread.sleep(100);
+        ledger.addEntry(createMessageWrittenToLedger("msg3"));
+        Thread.sleep(100);
```
Comment (315157973) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentMessageFinderTest.java:262]: Why we need sleep here, and what we are waiting for ? How can you ensure that it was finished after 100 ms, if we can't, there will be a flaky test

Comment (315157973) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentMessageFinderTest.java:262]: Why we need sleep here, and what we are waiting for ? How can you ensure that it was finished after 100 ms, if we can't, there will be a flaky test

**Review by HQebupt — APPROVED** [review]
LGTM

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentMessageFinderTest.java [hunk 2]
```diff
@@ -243,6 +243,43 @@ public void findEntryFailed(ManagedLedgerException exception, Optional<Position>
...
+
+        ManagedLedgerConfig config = new ManagedLedgerConfig();
+        config.setRetentionSizeInMB(10);
+        config.setMaxEntriesPerLedger(10);
+        config.setRetentionTime(1, TimeUnit.HOURS);
+        ManagedLedger ledger = factory.open(ledgerAndCursorName, config);
+        ManagedCursorImpl cursor = (ManagedCursorImpl) ledger.openCursor(ledgerAndCursorName);
+
+        ledger.addEntry(createMessageWrittenToLedger("msg1"));
+        Thread.sleep(100);
```
Comment (AnonHxy) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PersistentMessageFinderTest.java:258]: `ledger.addEntry` is a sync method, so  remove the sleep block is OK I think

**Review by AnonHxy — APPROVED** [review]
LGTM with minor comments

**Review by StevenLuMT — APPROVED** [review]
LGTM

**Comment by dave2wave** [comment]
You broke the build by checking in an empty file

**Pull request merged**